### PR TITLE
chore(factories) SubscriptionFactory - use LegalCitationProvider data

### LINKED ISSUE #273: chore: add dummy data for big cases
When I first got the system up and running and went to http://localhost:8888/big-cases/about/, there were no big cases being followed. I assumed that the initial dummy data would at least have one big case being followed so that something would show up on this page. Having some cases on the big cases page would let people see how cases show up on the page and that they are connected to Court Listener.

Having richer data will make it easier for developers to see how the models and information is related.  (Will especially help developers new to the project.)

At a minimum, I think `bootstrap-dev` should also:
1. create a channel Group for big cases: `name = 'big cases', is_big_cases=True`
2. set the Mastodon channel to be in the 'big cases' Group
3. add 2 of the existing Subscriptions to the Mastodon channel. Having more than 1 case lets a developer play with more variations.

I'm happy to do this. (I've already done the above just so I could see it on the big-cases page.)

Is there other minimal data that should be added?
Perhaps a Twitter channel should also be added? (And it should also be set to the 'big cases' Group)

### PR DESCRIPTION
**Pull request by weedySeaDragon**
for #273

Expand the `SubscriptionFactory` to use fake data generated by the Faker  `LegalCitationProvider` class. Also adds a docString for the post_generation method `channels()`

Depends on the chore/faker-citation-data-273 branch. This is the branch that adds `LegalCitationProvider`

### CODE REVIEW

**Comment by mlissner** [comment]
I've put this on @ERosendo's backlog for review in the next couple days. Thanks @weedySeaDragon.

File: bc/subscription/tests/factories.py [hunk 1]
```diff
@@ -13,13 +18,25 @@ class Meta:
...
+        @see https://factoryboy.readthedocs.io/en/stable/reference.html?highlight=post_generation#post-generation-hooks
+        @example:
+            some_channel = ChannelFactory.create()
+            SubscriptionFactory(channels=[channel_1, channel_2]) #
+            subscribes this to channel_1 and channel_2
+        """
         if not create or not extracted:
             return
-
+        print(f"extracted: {extracted}")
```
Comment (ERosendo) [inline bc/subscription/tests/factories.py:37]: I think we should remove this print statement.

Comment (weedySeaDragon) [reply bc/subscription/tests/factories.py:37]: Oops! Yes, I definitely should.

Comment (ERosendo) [reply bc/subscription/tests/factories.py]: Thanks!

**Pull request merged**